Update DBM's Postgres extended query protocol troubleshooting (#16285)
DataDog/documentation: content/en/database_monitoring/setup_postgres/troubleshooting.md

#### Postgres extended query protocol

> If a client is using the Postgres [extended query protocol][9] or prepared statements on Postgres version 12 or newer, enable the following beta feature in the [Postgres integration config][19].

Reviewer: This explanation jumps to a solution without explaining the background. Can you rewrite this for an audience who does not have any context about parameterized queries and explain plans.

> If a client is using the Postgres [extended query protocol][9] or prepared statements on Postgres version 12 or newer, enable the following beta feature in the [Postgres integration config](https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/data/conf.yaml.example).

Reviewer 2: Could you update this to use the reference link format, as mentioned in our contributing guidelines?
https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#text

Author: https://github.com/DataDog/documentation/pull/16285/commits/57a78b5915a5b3de946c0bcfd5432340cd25c4a8

```
query_samples:
  explain_parameterized_queries: true
  ...
```

> This enables the Agent to explain parameterized queries with generic values for the parameters. This may cause inaccuracies when the query references tables with partial indexes, or partition tables with varying indexes across child tables.

Reviewer: Drop this note. We'll explain the limitations of plans collected this way in-app.

> For versions older than Postgres 12, this feature is currently not supported and the Datadog Agent is unable to collect explain plans due to the separation of the parsed query and raw bind parameters. If the client provides an option to force using the simple query protocol, then turning that on enables the Datadog Agent to collect execution plans.

Reviewer 2: ```suggestion
For versions previous to Postgres 12 this feature is not supported, and the Datadog Agent is unable to collect explain plans due to the separation of the parsed query and raw bind parameters. If the client provides an option to force using the simple query protocol, the Datadog Agent is enabled to collect execution plans.
```
